sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveCatalogs.scala

```diff
@@ -50,10 +52,18 @@ class ResolveCatalogs(val catalogManager: CatalogManager)
 
     case CurrentNamespace =>
       ResolvedNamespace(currentCatalog, catalogManager.currentNamespace.toImmutableArraySeq)
-    case UnresolvedNamespace(Seq()) =>
-      ResolvedNamespace(currentCatalog, Seq.empty[String])
-    case UnresolvedNamespace(CatalogAndNamespace(catalog, ns)) =>
-      ResolvedNamespace(catalog, ns)
+    case UnresolvedNamespace(Seq(), _) =>
+      ResolvedNamespace(currentCatalog, Seq.empty[String], Map.empty)
```

Comment:
Let's add a function `def resolveNamespace(catalog: CatalogPlugin, ns: Seq[String]): ResolvedNamespace` and call it here.

Reply:
Nil is also a valid namespace in DS v2.

Reply:
Good point!